### ISSUE TITLE
disconnected private endpoint is not detected as config drift by resource type azurerm_private_endpoint

### ISSUE THREAD

**Issue by fotto** [hashicorp/terraform-provider-azurerm#23173]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.5.6

### AzureRM Provider Version

3.71.0

### Affected Resource(s)/Data Source(s)

azurerm_private_endpoint

### Terraform Configuration Files

```hcl
locals {
  bug_report_tags = {
    creator = "terraform"
    context = "tf-bug-report"
  }
  bug_name_suffix = "pep-tf-bug"
  bug_pep_settings = {
    # subnet to create deploy private endpoint in
    subnet_id = module.network.subnet_ids["restricted1"]
    # private DNS zone id where to create the private link DNS Alias
    dns_zone_id = data.azurerm_resources.shared_dns_zone_private_database.resources[0].id
  }
}

resource "random_password" "context_pwd" {
  length           = 16
  min_lower        = 2
  min_numeric      = 2
  min_upper        = 2
  special          = false
}

resource "azurerm_resource_group" "context_rg" {
  location = "westeurope"
  name     = "rg-${local.bug_name_suffix}"
  tags     = local.bug_report_tags
}

resource "azurerm_mssql_server" "context_sql" {
  name                         = "sql-${local.bug_name_suffix}"
  resource_group_name          = azurerm_resource_group.context_rg.name
  location                     = azurerm_resource_group.context_rg.location
  version                      = "12.0"

  administrator_login          = "local-admin"
  administrator_login_password = random_password.context_pwd.result

  minimum_tls_version          = "1.2"

  tags = local.bug_report_tags
}

resource "azurerm_private_endpoint" "pep" {
  name                = "pep-${local.bug_name_suffix}"
  resource_group_name = azurerm_resource_group.context_rg.name
  location            = azurerm_resource_group.context_rg.location
  subnet_id           = local.bug_pep_settings.subnet_id

  private_service_connection {
    name                              = "pep-${local.bug_name_suffix}"
    private_connection_resource_id    = azurerm_mssql_server.context_sql.id
    is_manual_connection              = false
    #request_message                   = "PL"
    subresource_names              = ["sqlServer"]
  }

  private_dns_zone_group {
    name = "default"
    private_dns_zone_ids = [local.bug_pep_settings.dns_zone_id]
  }

  tags = local.bug_report_tags
}
```

### Debug Output/Panic Output

```shell
> terraform plan

...resource list output removed for brevity...

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.
```

### Expected Behaviour

terraform plan/apply should detect a configuration drift (sub-resource "private_service_connection" is broken)
and suggest a change

### Actual Behaviour

no configuration change is detected (see output from "terraform plan" above)

### Steps to Reproduce

Prerequisites:

- VNet with subnet to create private endpoint in
- Private DNS zone for "privatelink.database.windows.net"

Steps to reproduce:

1. create resources (RG, SQL Server, Private Endpoint) with terraform code shown above ("terraform apply"). It will also create
   * PEP with Connection Status "Approved"
   * A-record for "sql-pep-tf-bug" is created in private zone "privatelink.database.windows.net"
   * network interface gets created (to hold the private IP)
2. action in Azure Portal
   * got to "SQL servers", select "sql-pep-tf-bug", go to Networking  -> Private access,
   * select the one private endpoint (pep-pep-tf-bug)  and click "Remove" (and confirm)
3. Checks in Azure Portal
   * Private Endpoint "pep-pep-tf-bug" will now show Connection status "Disconnected"
   * A-record for  "sql-pep-tf-bug" is not longer present in private zone "privatelink.database.windows.net"
4. Now run "terraform plan" => no change is detected ("terraform refresh") does not help

Workaround:  "taint" the azurerm_private_endpoint resource and re-run "terraform apply". That will re-created the PEP and fix the situation.

The main issue is the fact that the provider does not detect any configuration drift in that situation.

### References

#20776 seems to deal with a similar situation but with a completely different behavior of terraform

**Comment by sinbai** [comment]
Hi @fotto, thanks for opening this issue. I assume that this is by design. Because after step 2. of `Steps to reproduce`(select the one private endpoint (pep-pep-tf-bug) and click "Remove" (and confirm)) , the state of property [privateLinkServiceConnectionState](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/network/resource-manager/Microsoft.Network/stable/2023-05-01/privateEndpoint.json#L806C9-L806C9) in Azure API is changed from `Approved` to `Disconnected` (Terraform provider relies on Azure API to manage Azure resources). Per the description of `privateLinkServiceConnectionState`, this is a read-only property, However, provider doesn't not manage the read-only property(this is by design), So provider could not detect the state has changed(as you see the symptom: provider does not detect any configuration drift in that situation).

**Comment by fotto** [comment]
Hi @sinbai thanks for your explanation. 

The problem here is that terraform creates the impression that everything is as it should be. But actually the resource is broken and only a re-deployment can fix it.

Ideally the provider would respond with something like:  
   azurerm_private_endpoint "pep-pep-tf-bug" must be recreated 
and would suggest destroy+create.

Would it be possible to include the ConnectionState attribute into the state (as immutable attribute => forces re-deployment) where the desired value is always "Approved"  (but not sent on creation/update)?

**Comment by jonathan-vogel-siemens** [comment]
@fotto i agree. This is exactly how the provider should behave also in my opinion.
